### PR TITLE
e2e: Add test for https://github.com/rectorphp/rector/issues/7231

### PR DESCRIPTION
**Pull request by jtojnar**

### CODE REVIEW

**Comment by samsonasik** [comment]
You need to register new e2e dir to :

https://github.com/rectorphp/rector-src/blob/2be958b4e4db947c63777443eafb886274a5c7c7/.github/workflows/e2e.yaml#L25

File: e2e/removed-and-added-files-collector/rector.php [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+require_once __DIR__ . '/TestRector.php';
+
+return static function (Rector\Config\RectorConfig $rectorConfig): void {
+    $rectorConfig->rule(TestRector::class);
+    $rectorConfig->paths([
+        __DIR__ . '/src',
+    ]);
+
+    // TODO: Make it run in parallel.
+    $rectorConfig->disableParallel();
```
Comment (jtojnar) [inline e2e/removed-and-added-files-collector/rector.php:16]: To reproduce the failure, remove this line.

File: e2e/removed-and-added-files-collector/expected-output.diff [hunk 2]
```diff
@@ -0,0 +1,6 @@
+! [NOTE] File "./removed-and-added-files-collector/test" will be added
```
Comment (jtojnar) [inline e2e/removed-and-added-files-collector/expected-output.diff:1]: No idea why CI messes it up like this:

```diff
@@ @@
-! [NOTE] File
- !        "./removed-and-added-files-c
- !        ollector/test" will be added
+! [NOTE] File "./removed-and-added-files-collector/test" will be added
```

**Comment by samsonasik** [comment]
`$this->removedAndAddedFilesProcessor->run($configuration)` seems missing in parallel process.

**Pull request closed without merging**